### 1. 题目：有四个数字：1、3、5、7，能组成多少个互不相同且无重复数字的三位数？各是多少？

In [1]:
for i in range(1,8,2):
    for j in range(1,8,2):
        for k in range(1,8,2):
            if( i != k ) and (i != j) and (j != k):
                print(i*100+j*10+k)

135
137
153
157
173
175
315
317
351
357
371
375
513
517
531
537
571
573
713
715
731
735
751
753


### 2 . 题目：企业发放的奖金根据利润提成。利润计算方法如下：
* 低于或等于10万元时，奖金可提10%；  
* 利润高于10万元，低于20万元时，低于10万元的部分按10%提成，高于10万元的部分，可提成7.5%；  
* 20万到40万之间时，高于20万元的部分，可提成5%；  
* 40万到60万之间时高于40万元的部分，可提成3%；  
* 60万到100万之间时，高于60万元的部分，可提成1.5%，高于100万元时，超过100万元的部分按1%提成。  
#### 利用input函数从键盘输入当月利润I，求应发放奖金总数？ 

In [2]:
i = float(input('请输入净利润(万元):'))
arr = [100,60,40,20,10,0]
rat = [0.01,0.015,0.03,0.05,0.075,0.1]
r = 0
for idx in range(0,6):
    if i>arr[idx]:
        r+=(i-arr[idx])*rat[idx]
        print((i-arr[idx])*rat[idx])
        i=arr[idx]
print("总奖金:", r)

请输入净利润(万元):100
0.6
0.6
1.0
0.75
1.0
总奖金: 3.95


### 3. 题目：输入某年某月某日，判断这一天是这一年的第几天？

In [3]:
year = int(input('year:\n'))
month = int(input('month:\n'))
day = int(input('day:\n'))
 
months = (0,31,59,90,120,151,181,212,243,273,304,334)
if 0 < month <= 12:
    sum = months[month - 1]
else:
    print('data error')
sum += day
leap = 0
if (year % 400 == 0) or ((year % 4 == 0) and (year % 100 != 0)):
    leap = 1
if (leap == 1) and (month > 2):
    sum += 1
print('it is the %dth day.' % sum)

year:
2018
month:
11
day:
2
it is the 306th day.


### 4. 题目：输入三个整数x,y,z，请把这三个数由小到大输出。

In [4]:
l = []
for i in range(3):
    x = int(input('integer:\n'))
    l.append(x)
l.sort()
print(l)

integer:
4
integer:
9
integer:
9
[4, 9, 9]


### 5. 题目：构造斐波那契数列，并输出数列中的第10个数。

In [5]:
def fib(n):
    a,b = 1,1
    for i in range(n-1):
        a,b = b,a+b
    return a
 
# 输出了第10个斐波那契数列
print(fib(10))

55


In [6]:
# 使用递归
def fib(n):
    if n==1 or n==2:
        return 1
    return fib(n-1)+fib(n-2)
 
# 输出了第10个斐波那契数列
print(fib(10))

55


### 6. 题目：输出 9*9 乘法口诀表。

In [15]:
for i in range(1, 10):
    
    for j in range(1, i+1):
        print("%d * %d = %2d" % (i, j, i*j), end="; ")
    print("\b\b")

1 * 1 =  1; 
2 * 1 =  2; 2 * 2 =  4; 
3 * 1 =  3; 3 * 2 =  6; 3 * 3 =  9; 
4 * 1 =  4; 4 * 2 =  8; 4 * 3 = 12; 4 * 4 = 16; 
5 * 1 =  5; 5 * 2 = 10; 5 * 3 = 15; 5 * 4 = 20; 5 * 5 = 25; 
6 * 1 =  6; 6 * 2 = 12; 6 * 3 = 18; 6 * 4 = 24; 6 * 5 = 30; 6 * 6 = 36; 
7 * 1 =  7; 7 * 2 = 14; 7 * 3 = 21; 7 * 4 = 28; 7 * 5 = 35; 7 * 6 = 42; 7 * 7 = 49; 
8 * 1 =  8; 8 * 2 = 16; 8 * 3 = 24; 8 * 4 = 32; 8 * 5 = 40; 8 * 6 = 48; 8 * 7 = 56; 8 * 8 = 64; 
9 * 1 =  9; 9 * 2 = 18; 9 * 3 = 27; 9 * 4 = 36; 9 * 5 = 45; 9 * 6 = 54; 9 * 7 = 63; 9 * 8 = 72; 9 * 9 = 81; 


### 7. 题目：暂停一秒输出，并格式化当前时间。

In [16]:
import time
 
print(time.strftime('%Y-%m-%d %H:%M:%S',time.localtime(time.time())))
 
# 暂停一秒
time.sleep(1)
 
print(time.strftime('%Y-%m-%d %H:%M:%S',time.localtime(time.time())))

2021-03-06 18:48:34
2021-03-06 18:48:35


### 8. 题目：判断101-200之间有多少个素数，并输出所有素数。

In [17]:
count = 0
leap = 1
from math import sqrt
for m in range(101,201):
    k = int(sqrt(m + 1))
    for i in range(2,k + 1):
        if m % i == 0:
            leap = 0
            break
    if leap == 1:
        print('%-4d' % m)
        count += 1
        if count % 10 == 0:
            print()
    leap = 1
print('The total is %d' % count)

101 
103 
107 
109 
113 
127 
131 
137 
139 
149 

151 
157 
163 
167 
173 
179 
181 
191 
193 
197 

199 
The total is 21


### 9. 题目：打印出所有的"水仙花数"，所谓"水仙花数"是指一个三位数，其各位数字立方和等于该数本身。例如：153是一个"水仙花数"，因为153=1的三次方＋5的三次方＋3的三次方。

In [19]:
for n in range(100,1000):
    i = int(n / 100)
    j = int(n / 10 % 10)
    k = int(n % 10)
    if n == i ** 3 + j ** 3 + k ** 3:
        print(n)

153
370
371
407


In [20]:
print(i,j,k)

9 9 9


### 10. 题目：将一个正整数分解质因数。例如：输入90，则打印出90=2 * 2 * 2 * 3 * 5。

程序分析：对n进行分解质因数，应先找到一个最小的质数k，然后按下述步骤完成：
(1)如果这个质数恰等于n，则说明分解质因数的过程已经结束，打印出即可。
(2)如果n<>k，但n能被k整除，则应打印出k的值，并用n除以k的商,作为新的正整数你n,重复执行第一步。
(3)如果n不能被k整除，则用k+1作为k的值,重复执行第一步。

程序源代码：

In [22]:
def reduceNum(n):
    print('{} = '.format(n),end='')
    if not isinstance(n, int) or n <= 0 :
        print('请输入一个正确的数字 !')
        exit(0)
    elif n in [1] :
        print('{}'.format(n))
    while n not in [1] : # 循环保证递归
        for index in range(2, int(n) + 1) :
            if n % index == 0:
                n /= index # n 等于 n/index
                if n == 1: 
                    print(index,end='')
                else : # index 一定是素数
                    print('{} * '.format(index),end='')
                break
reduceNum(1)
reduceNum(100)

1 = 1
100 = 2 * 2 * 5 * 5